packages/go/cypher/frontend/query.go

```diff
@@ -363,6 +363,7 @@ func (s *SinglePartQueryVisitor) EnterOC_UpdatingClause(ctx *parser.OC_UpdatingC
 }
 
 func (s *SinglePartQueryVisitor) ExitOC_UpdatingClause(ctx *parser.OC_UpdatingClauseContext) {
+	s.ctx.HasMutation = true
```

Comment:
just curious, is there a reason why hasMutation is set to true on exit as opposed to on enter?

Reply:
Exit was where the clause is actually added to the query struct so it felt more appropriate.

Also the filtering system relies on overriding the Enter variants and that caused some pain points so this ended up being the spot we tossed the flag. Not opposed to another area if one is a better fit / anyone has strong opinions though!